### PR TITLE
Hack the Helm install client and improve the helm chart

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
1. Create a factory to convert a controller client to a kube CLI client so the helm install would work in cluster
2. Move the cert-manager to be a dependency chart
3. Improve the Chart details like randomize some names

### CODE REVIEW

File: api/types/types.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package types
 
 const (
-	DefaultOAMNS               = "oam-system"
-	DefaultOAMReleaseName      = "vela-core"
+	DefaultOAMNS               = "vela-system"
```
Comment (wonderflow) [inline api/types/types.go:4]: Why change the namespace

File: api/types/types.go [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+	DefaultOAMReleaseName      = "kube"
```
Comment (wonderflow) [inline api/types/types.go:5]: Why name it kube

**Review by wonderflow — APPROVED** [review]
Generally LGTM

File: api/types/types.go (hunk 2 above)
Comment (zzxwill) [inline api/types/types.go:5]: I prefer the old OAM release as users might `kube` might be mixed up with kube-xxx releases:) And I like the new OAM namespace.

File: api/types/types.go (hunk 1 above)
Comment (ryanzhang-oss) [reply api/types/types.go:4]: Is vela part of "oam-system"

File: api/types/types.go (hunk 2 above)
Comment (ryanzhang-oss) [reply api/types/types.go:5]: Isn't the official name "kubeVela"?

**Comment by ryanzhang-oss** [comment]
btw, I tried to move the cert-manager to be a dependency and it works if I just use helm. It doesn't work when I tried to incorporate that into `vela install`

**Pull request merged**